File: tests/algorithms/x/OHP/strings/snake_case_to_camel_pascal_case.php
<?php
error_reporting(E_ALL & ~E_DEPRECATED);
ini_set('memory_limit', '-1');
$now_seed = 0;
$now_seeded = false;
$s = getenv('MOCHI_NOW_SEED');
if ($s !== false && $s !== '') {
    $now_seed = intval($s);
    $now_seeded = true;
}
function _now() {
    global $now_seed, $now_seeded;
    if ($now_seeded) {
        $now_seed = ($now_seed * 1664525 + 1013904223) % 2147483647;
        return $now_seed;
    }
    return hrtime(true);
}
function _append($arr, $x) {
    $arr[] = $x;
    return $arr;
}
$__start_mem = memory_get_usage();
$__start = _now();
  function mochi_split($s, $sep) {
  $res = [];
  $current = '';
  $i = 0;
  while ($i < strlen($s)) {
  $ch = substr($s, $i, $i + 1 - $i);
  if ($ch == $sep) {
  $res = _append($res, $current);
  $current = '';
} else {
  $current = $current . $ch;
}
  $i = $i + 1;
};
  $res = _append($res, $current);
  return $res;
};
  function capitalize($word) {
  if (strlen($word) == 0) {
  return '';
}
  $first = strtoupper(substr($word, 0, 1));
  $rest = substr($word, 1, strlen($word) - 1);
  return $first . $rest;
};
  function snake_to_camel_case($input_str, $use_pascal) {
  $words = mochi_split($input_str, '_');
  $result = '';
  $index = 0;
  if (!$use_pascal) {
  if (count($words) > 0) {
  $result = $words[0];
  $index = 1;
};
}
  while ($index < count($words)) {
  $word = $words[$index];
  $result = $result . capitalize($word);
  $index = $index + 1;
};
  return $result;
};
  echo rtrim(snake_to_camel_case('some_random_string', false)), PHP_EOL;
  echo rtrim(snake_to_camel_case('some_random_string', true)), PHP_EOL;
  echo rtrim(snake_to_camel_case('some_random_string_with_numbers_123', false)), PHP_EOL;
  echo rtrim(snake_to_camel_case('some_random_string_with_numbers_123', true)), PHP_EOL;
$__end = _now();
$__end_mem = memory_get_peak_usage(true);
$__duration = max(1, intdiv($__end - $__start, 1000));
$__mem_diff = max(0, $__end_mem - $__start_mem);
$__bench = ["duration_us" => $__duration, "memory_bytes" => $__mem_diff, "name" => "main"];
$__j = json_encode($__bench, 128);
$__j = str_replace("    ", "  ", $__j);
echo $__j, PHP_EOL;

File: tests/algorithms/transpiler/FS/strings/snake_case_to_camel_pascal_case.fs
// Generated 2025-08-11 15:32 +0700

exception Return
let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let _substring (s:string) (start:int) (finish:int) =
    let len = String.length s
    let mutable st = if start < 0 then len + start else start
    let mutable en = if finish < 0 then len + finish else finish
    if st < 0 then st <- 0
    if st > len then st <- len
    if en > len then en <- len
    if st > en then st <- en
    s.Substring(st, en - st)

let _idx (arr:'a array) (i:int) : 'a =
    if not (obj.ReferenceEquals(arr, null)) && i >= 0 && i < arr.Length then arr.[i] else Unchecked.defaultof<'a>
let __bench_start = _now()
let __mem_start = System.GC.GetTotalMemory(true)
let rec split (s: string) (sep: string) =
    let mutable __ret : string array = Unchecked.defaultof<string array>
    let mutable s = s
    let mutable sep = sep
    try
        let mutable res: string array = Array.empty<string>
        let mutable current: string = ""
        let mutable i: int = 0
        while i < (String.length (s)) do
            let ch: string = _substring s i (i + 1)
            if ch = sep then
                res <- Array.append res [|current|]
                current <- ""
            else
                current <- current + ch
            i <- i + 1
        res <- Array.append res [|current|]
        __ret <- res
        raise Return
        __ret
    with
        | Return -> __ret
let rec capitalize (word: string) =
    let mutable __ret : string = Unchecked.defaultof<string>
    let mutable word = word
    try
        if (String.length (word)) = 0 then
            __ret <- ""
            raise Return
        let first: string = (_substring word 0 1).ToUpper()
        let rest: string = _substring word 1 (String.length (word))
        __ret <- first + rest
        raise Return
        __ret
    with
        | Return -> __ret
let rec snake_to_camel_case (input_str: string) (use_pascal: bool) =
    let mutable __ret : string = Unchecked.defaultof<string>
    let mutable input_str = input_str
    let mutable use_pascal = use_pascal
    try
        let words: string array = split (input_str) ("_")
        let mutable result: string = ""
        let mutable index: int = 0
        if not use_pascal then
            if (Seq.length (words)) > 0 then
                result <- _idx words (int 0)
                index <- 1
        while index < (Seq.length (words)) do
            let word: string = _idx words (int index)
            result <- result + (capitalize (word))
            index <- index + 1
        __ret <- result
        raise Return
        __ret
    with
        | Return -> __ret
printfn "%s" (snake_to_camel_case ("some_random_string") (false))
printfn "%s" (snake_to_camel_case ("some_random_string") (true))
printfn "%s" (snake_to_camel_case ("some_random_string_with_numbers_123") (false))
printfn "%s" (snake_to_camel_case ("some_random_string_with_numbers_123") (true))
let __bench_end = _now()
let __mem_end = System.GC.GetTotalMemory(true)
printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)
